mediatek: filogic: support openwrt,netdev-name; set labels for sfp ports on bpi-r3 and bpi-r4

jonasjelonek opened the pull request:
The LAN and WAN ports, which are connected to the switch chips on Banana Pi R3 and R4, are named properly in the device tree like `wan` and `lanX` (and `sfp2` on R3). However, the other SFP slots are directly connected to the SoC and have their own Ethernet nodes in the device tree. Labelling these is afaik not supported upstream and thus they are automatically named like `eth1` and `eth2` (which is even in bad order on the R4 - `eth2` is first/left SFP port and `eth1` is the second/right).

Thus, set the labels in the `gmac` nodes in the device trees of Banana Pi R3 and R4 with labels that match the labels printed on the PCB/official metal case. The labels are picked up and set by the `04_set_netdev_labels` script/hook in `filogic` subtarget.
On R3, the first SFP port is renamed from `eth1` to `sfp1`. On R4, the first SFP port is renamed from `eth2` to `sfp-wan` and the second is renamed from `eth1` to `sfp-lan`.
Also bump the `compat_version` to indicate that the configuration is incompatible with this change.

This also renames the port of the R4 PoE version, however there is no reference for the name I've chosen since there doesn't seem to be a dedicated label on the board or another faceplate for the metal case (at least I haven't found it).

ping @dangowrt 

EDIT: Due to recent commit 5695267847 I also added another commit which adds support for our own custom dts property `openwrt,netdev-name` for renaming interfaces, as suggested upstream [1].

[1] https://lore.kernel.org/netdev/20240709124503.pubki5nwjfbedhhy@skbuf/

jonasjelonek committed:
mediatek: filogic: bpi-r3: set netdev-name for sfp1 port

Sets openwrt,netdev-name for the gmac1 node in the dts of BPI-R3, which
corresponds to the sfp1 slot, to have a proper naming and match the
label on the official BPI-R3 metal case. This renames the port from eth1
to sfp1.
The COMPAT_VERSION is increased to denote that configuration has to be
adjusted manually.

Signed-off-by: Jonas Jelonek <jelonek.jonas@gmail.com>

jonasjelonek committed:
mediatek: filogic: bpi-r4: set netdev-name for sfp ports

Sets openwrt,netdev-name for the gmac nodes in the dts of BPI-R4 which
correspond to the two sfp slots. By default they are automatically
named as eth1 and eth2 in bad order, however 'SFP1-WAN' and 'SFP2-LAN'
are printed on the PCB and the official metal case has labels 'SFP-WAN'
and 'SFP-LAN'. Thus, label the ports accordingly to match the
board/case labels.
The COMPAT_VERSION is increased to denote that configuration has to be
adjusted manually.

Signed-off-by: Jonas Jelonek <jelonek.jonas@gmail.com>

jonasjelonek commented:
@robimarko Just saw your commit 5695267847. Is this now the recommended property to use for device-specific interface renaming? My patches still rely on the previous property `label`.

robimarko commented:
Either will work since if its standalone ethernet controller then kernel does not use the label property, though we probably will eventually want to move to our custom property to indicate that is our downstream solution and in case that kernel decides to do its thing we will not be affected

jonasjelonek committed:
mediatek: filogic: support openwrt,netdev-name for renaming interfaces

Add support in filogic subtarget for our own custom property
`openwrt,netdev-name` in the device tree instead of `label` for
renaming interfaces. This was suggested upstream to avoid potential
conflicts [1].

[1] https://lore.kernel.org/netdev/20240709124503.pubki5nwjfbedhhy@skbuf/

Signed-off-by: Jonas Jelonek <jelonek.jonas@gmail.com>

jonasjelonek commented:
> Either will work since if its standalone ethernet controller then kernel does not use the label property, though we probably will eventually want to move to our custom property to indicate that is our downstream solution and in case that kernel decides to do its thing we will not be affected

could you please provide me with the link where upstream suggested this? Can't find it in your commit or PR.

jonasjelonek commented:
I just added a commit to switch filogic to the custom property. I couldn't find devices that rely on that renaming script, so only the script is adjusted. However, I may have overseen something, so please raise your hand here in that case.

robimarko commented:
No, you dont need to update the script at all, just rebase your branch on top of master as it already supports the custom `openwrt,netdev-name` property

jonasjelonek commented:
@robimarko I'm a bit confused now. If I remove the commit that adjusts the script, the renaming doesn't work anymore.
Your commit adjusts the script for mt7621, but not for filogic or is that expected to work anyway?

robimarko commented:
Ah, you are right, this is filogic, this is what happens when you are doing 5 things at the same time.

I would avoid changing the script to only accept our custom property, and would do the same thing I did and add it as a second supported method after label

robimarko commented:
Just a nit pick, you did not update the commit title and description, you are adding support for it, not switching to only it

jonasjelonek commented:
Missed that, thanks for the hint. Will fix that in a moment.

jonasjelonek commented:
ping @dangowrt 
since you added support for these devices, what do you think about my changes?

jonasjelonek commented:
merged with d4d6c48b6e 84fc59c0d5 cd8dcfef37 
thanks @aparcar
